### PR TITLE
NDRS-895: prepare for upgrading delta nodes

### PR DESCRIPTION
**Pull request by Fraser999**
Ref: https://casperlabs.atlassian.net/browse/NDRS-895

This PR brings over almost all commits from master to this feature branch in order to allow for a potential upgrade to nodes currently running on the delta network.

The primary commit which is different to master is [f5b1b6f](https://github.com/CasperLabs/casper-node/commit/f5b1b6fc57701e33b555e51451ca05e093dd501f) which reverts the structure of `Block` to match that on the delta nodes.  This would be the only part of this PR worth reviewing IMO.

### CODE REVIEW

**Review by fizyk20 — APPROVED** [review]
The commits tagged `NDRS-895` (so the main commit mentioned in the description and a few others) look fine to me. The others, I assume, are just directly ported from `master`.

File: node/src/types/block.rs [hunk 1]
```diff
@@ -582,7 +597,7 @@ pub struct BlockHeader {
     transfer_hashes: Vec<DeployHash>,
     random_bit: bool,
     accumulated_seed: Digest,
-    era_end: Option<EraEnd>,
```
Comment (goral09) [inline node/src/types/block.rs:585]: This changes the structure of a `BlockHeader`, I thought we didn't want to do it for delta.

**Review by goral09 — CHANGES_REQUESTED** [review]
Are we sure this change to the `BlockHeader` is safe? Wouldn't serialization change? And definitely we would be changing the SSE events.

File: node/src/types/block.rs (hunk 1 above)
Comment (goral09) [reply node/src/types/block.rs:585]: [`4375f11` (#943)](https://github.com/CasperLabs/casper-node/pull/943/commits/4375f11ef896eaa788492e5ae03179b80c79846f) also changes the structure

Comment (fizyk20) [reply node/src/types/block.rs:585]: I believe this _reverts_ a change that was on `master`, but not on the release branch... doesn't it?

Comment (fizyk20) [reply node/src/types/block.rs:585]: Although... it looks like it was called `EraEnd` on the original release branch? So maybe the name shouldn't change.

Comment (goral09) [reply node/src/types/block.rs:585]: This PR is against the old release (Delta) branch so we will see the changes to the delta.

Comment (Fraser999) [reply node/src/types/block.rs:585]: Yes, exactly

**Comment by Fraser999** [comment]
bors merge

**Comment by Fraser999** [comment]
bors merge

**Pull request merged**